File: csharp/src/PrintDate/PrintDate.cs
namespace PrintDate;

public class PrintDate
{
    private readonly ICalendar _calendar;
    private readonly IPrinter _printer;

    public PrintDate(ICalendar calendar, IPrinter printer)
    {
        _calendar = calendar;
        _printer = printer;
    }

    public void PrintCurrentDate()
    {
        var line = _calendar.Today().ToString();
        _printer.PrintLine(line);
    }
}

File: go/printdate/printdate.go
package printdate

type PrintDate interface {
	PrintCurrentDate()
}

func NewPrintDate(calendar Calendar, printer Printer) PrintDate {
	return printDate{calendar: calendar, printer: printer}
}

type printDate struct {
	calendar Calendar
	printer  Printer
}

func (printDate printDate) PrintCurrentDate() {
	today := printDate.calendar.today()
	printDate.printer.printLine(today.String())
}
